org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/BuildWorkspaceHandlerTest.java

```diff
@@ -31,34 +33,35 @@
 	private JavaClientConnection javaClient = new JavaClientConnection(client);
 
 	private BuildWorkspaceHandler handler;
-	private IFile file;
+	private IProject project;
 
 	@Before
 	public void setUp() throws Exception {
-		handler = new BuildWorkspaceHandler(javaClient, projectsManager, new WorkspaceDiagnosticsHandler(javaClient, projectsManager));
+		handler = new BuildWorkspaceHandler(javaClient, projectsManager);
 		importProjects("maven/salut2");
-		file = linkFilesToDefaultProject("singlefile/Single.java");
+		project = ResourcesPlugin.getWorkspace().getRoot().getProject("salut2");
 	}
 
 	@Test
 	public void testSucceedCase() throws Exception {
 		BuildWorkspaceStatus result = handler.buildWorkspace(false, monitor);
-		waitForBackgroundJobs();
 		assertTrue(String.format("BuildWorkspaceStatus is: %s.", result.toString()), result == BuildWorkspaceStatus.SUCCEED);
 	}
 
 	@Test
 	public void testFailedCase() throws Exception {
-		String codeWithError = "public class Single2 {\n" +
+		//@formatter:off
+		String codeWithError = "package foo;\n" +
+				"	public class Single2 {\n" +
 				"	public static void main(String[] args){\n"+
 				"		int ss = 1;;\n"+
 				"	}\n"+
 				"}";
+		//@formatter:on
+		IFile file = project.getFile("src/main/java/foo/Bar.java");
```

Comment:
I think the purpose of this test was to check standalone java files would cause the build to fail in case of an error